app/controllers/posts_controller.rb

```diff
@@ -1,86 +1,61 @@
 class PostsController < ApplicationController
   before_action :authenticate_user,except: [:index,:show]
 
-
   before_action :find_post,except: [:index,:new,:create]
 
 
-
-
-  def find_post
-    @post = Post.find params[:id]
```

Comment:
You don't need this line because you have  this one 'before_action :find_post,except: [:index,:new,:create]'